src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -61,6 +61,17 @@ namespace fastrtps {
 namespace rtps {
 
 
+static void add_statistics_sent_submessage(
```

Comment:
I think this method should be either:
- a static method of `RTPSWriter` to avoid a StatelessWriter redefinition of the function.
- a setter in the `CacheChange_t` struct.

Reply:
I made both `add_statistics_sent_submessage` and `send_data_or_fragments` part of RTPSWriter.